lang/java/java.talon

```diff
@@ -0,0 +1,56 @@
+mode: user.java
+mode: user.auto_lang
+and code.language: java
+-
+tag(): user.code_operators
+tag(): user.code_comment
+tag(): user.code_block_comment
+tag(): user.code_generic
+
+action(user.code_state_import):
+    insert("import ")
+
+# Types Commands
+type {user.code_types}: 
+    insert(code_types)
+    key("space")
+
+type ({user.java_common_classes}|{user.java_generic_data_structures}): 
+    insert(java_common_classes or java_generic_data_structures)
+    key("space")
+
+boxed [type] {user.java_boxed_types}: 
+    insert(user.java_boxed_types)
+    key("space")
+
+generic [type] {user.java_generic_data_structures}:
+    insert(java_generic_data_structures)
+    insert("<>")
+    key("left")
+
+# Arrays
+type {user.code_types} array:
+    insert(user.code_types)
+    user.code_operator_subscript()
+
+type ({user.java_common_classes}|{user.java_generic_data_structures}) array:
+    insert(java_common_classes or java_generic_data_structures)
+    user.code_operator_subscript()    
+
+[state] {user.java_access_modifiers}: 
```

Comment:
C#, Python, and javascript all use 'pro static funky' type things to make functions (and assume a 'void' return type in the case of C#). Maybe the java stuff should work the same way as them? There's no direct support for class access modifiers in the C# implementation either.

I guess the difference would be between saying "public static boxed integer camel my fancy method" -> "public static Integer myFancyMethod|" and "pub static my fancy method" -> "public static void myFancyMethod(|)"and then having to go back and change the void to an integer in the second case.